hal/src/electron/modem/dns_client.cpp

```diff
@@ -162,13 +165,21 @@ class UdpSocket {
     }
 
     int recvFrom(MDM_IP* addr, int* port, char* data, size_t size) {
+        // See if any data is waiting for us
+        int result = electronMDM.socketReadable(sock_);
+        if (result <= 0) {
```

Comment:
Shouldn't the negative errors be propagated as-is instead?

Reply:
This mimics the same operation as `socket_receivefrom()` in `socket_hal.cpp`.  Even though `socketReadable()` returns -1 most of the time when there is no data, we interpret that as no data, same as 0.  We wouldn't want to detect -1 and close the socket either.  We need to return 0 specifically to keep this code from bailing out:
```c++
ret = recvDnsResponse(&sock, &q);
if (ret < 0) {
    return ret;
}
```

Reply:
> Even though `socketReadable()` returns -1 most of the time when there is no data

I thought it returned `0` if there is no data and `-1` in case of errors last time I looked at its code, so that's why I suggested short-circuiting on negative values.

Reply:
I'm going to mark this one resolved, please let me know if you think there is more to do here.

Reply:
I still don't understand why we're hiding a potential error here :man_shrugging: